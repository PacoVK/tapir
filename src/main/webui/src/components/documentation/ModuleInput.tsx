import React from "react";
import { Badge, Box, Container, Stack, Typography } from "@mui/material";
import { ModuleInputProps } from "../../types";
import SyntaxHighlighter from "react-syntax-highlighter/dist/cjs/light";
import { ghcolors as theme } from "react-syntax-highlighter/dist/esm/styles/prism";

const ModuleInput = (props: ModuleInputProps) => {
  const { inputs } = props;

  const renderDefaultValue = (defaultValue: any) => {
    if (typeof defaultValue === "object") {
      return JSON.stringify(defaultValue);
    } else {
      return defaultValue;
    }
  };

  return (
    <Stack justifyContent="center" alignItems="left" spacing={2}>
      {inputs && inputs.length > 0 ? (
        inputs.map((input, index) => {
          return (
            <Box key={`module-input-${index}`}>
              <Typography>
                <Badge
                  color="secondary"
                  variant="dot"
                  invisible={!input.required}
                >
                  <strong>{input.name}</strong>
                </Badge>
              </Typography>
              <Container sx={{ ml: "inherit" }} maxWidth="sm">
                <Typography>
                  {input.description ? input.description : null}
                </Typography>
                <SyntaxHighlighter language="hcl" style={theme}>
                  {`${input.type} ${
                    input.default
                      ? `= ${renderDefaultValue(input.default)}`
                      : ""
                  }`}
                </SyntaxHighlighter>
              </Container>
            </Box>
          );
        })
      ) : (
        <Typography>This module has no inputs</Typography>
      )}
    </Stack>
  );
};

export default ModuleInput;
