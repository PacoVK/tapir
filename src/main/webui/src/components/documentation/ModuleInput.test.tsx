import { render } from "@testing-library/react";
import ModuleInput from "./ModuleInput";

describe("<ModuleInput /> spec", () => {
  it("renders the ModuleInput", () => {
    const view = render(
      <ModuleInput
        inputs={[
          {
            name: "repositories",
            type: "map(object({force_delete : optional(bool, false)\n    tags : optional(map(any))\n  }))",
            description: "List of repositories to create",
            required: false,
          },
            {
                name: "cidr_blocks",
                type: "string",
                description: "CIDR block for the VPC",
                required: true,
                default: "10.0.0.0/8",
            },
        ]}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
