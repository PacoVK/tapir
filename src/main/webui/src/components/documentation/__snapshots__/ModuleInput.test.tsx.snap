// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleInput /> spec renders the ModuleInput 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-c08hs3-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            <span
              class="MuiBadge-root css-1c32n2y-MuiBadge-root"
            >
              <strong>
                repositories
              </strong>
              <span
                class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-rt4zii-MuiBadge-badge"
              />
            </span>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              List of repositories to create
            </p>
            <pre
              style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border: 1px solid #dddddd; background-color: white;"
            >
              <code
                class="language-hcl"
                style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none;"
              >
                <span>
                  map(object({force_delete : optional(bool, false)

                </span>
                    tags : optional(map(any))

                  }))
              </code>
            </pre>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-c08hs3-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <span
            class="MuiBadge-root css-1c32n2y-MuiBadge-root"
          >
            <strong>
              repositories
            </strong>
            <span
              class="MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-rt4zii-MuiBadge-badge"
            />
          </span>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-10a3hyv-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            List of repositories to create
          </p>
          <pre
            style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border: 1px solid #dddddd; background-color: white;"
          >
            <code
              class="language-hcl"
              style="color: rgb(57, 58, 52); font-family: \\"Consolas\\", \\"Bitstream Vera Sans Mono\\", \\"Courier New\\", Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; font-size: .9em; line-height: 1.2em; tab-size: 4; hyphens: none;"
            >
              <span>
                map(object({force_delete : optional(bool, false)

              </span>
                  tags : optional(map(any))

                }))
            </code>
          </pre>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
